src/utils/telemetryEventQueue.ts

```diff
@@ -1,5 +1,4 @@
-import { TelemetryEvent } from './TelemetryEvent';
-import { Reporter } from './Reporter';
+import { TelemetryEvent } from '../interfaces/telemetryEvent';
 
 let queue: TelemetryEvent[] | undefined;
```

Comment:
I see that no limits being put on to the no.of elements in the queue. Wouldn't this case an issue if user never opts for telemetry but we keep enqueuing the events into in memory queue?

Reply:
indeed, should make it FIFO

Reply:
Added the logic to limit the queue during addEvent.

Reply:
FIFO will work with shift logic

Reply:
@fbricon @arajkumar here is the logic. https://github.com/redhat-developer/vscode-commons/pull/3/commits/99053188c0ea2fb027fc11604a638fae43e45e67
we can close on this if it is acceptable.